### DISCUSSION TITLE
Rewrites not working in Next 12.3.1

### DISCUSSION THREAD

**Discussion by adityar15** [vercel/next.js#41356] [Help]
This is what I have in my next.config.js

```
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
  async rewrites(){
    return[
      {
        source: "/",
        destination: "/about",
      }
    ]
  }
}

module.exports = nextConfig

```

Whenever I visit localhost:3000, I see the content of index.js rather than seeing the content of about.js. Am I doing something wrong here or is there any issue with Next's latest version as it used to work fine before?

Thanks :)

**Comment by icyJoseph** [comment]
Try with this:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
  async rewrites() {
    return {
      beforeFiles: [{ source: "/", destination: "/blog" }],
    };
  },
};

module.exports = nextConfig;
```

Though yours should work, not sure what's up there 😅

**Reply by adityar15** [reply]
Hi, thanks for the reply. I tried everything possible. Cleared up the cache, and tried in incognito mode. Not sure either what's wrong.

**Comment by samithaf** [comment]
Can you try like this?

`source: '{/}?',`

**Reply by icyJoseph** [reply]
The thing is that the docs document exactly the way OP is doing it, so it should work, unless we are missing some more context.

**Comment by adityar15** [comment]
Alright, I implemented the solution provided by @icyJoseph
```
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
  async rewrites() {
    return {
      beforeFiles: [{ source: "/", destination: "/blog" }],
    };
  },
};

module.exports = nextConfig;
```

Thanks for the solution. For some reason, it works.
